src/analytics.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+
+import fs from 'fs-extra'
+import { default as OS } from 'os'
+import type Config from 'cli-engine-config'
+import HTTP from 'cli-engine-command/lib/http'
+import Netrc from 'netrc-parser'
+import Output from 'cli-engine-command/lib/output'
+import URI from 'url'
+
+export default class AnalyticsCommand {
+  command: string
+  plugin: string
+  pluginVersion: string
+  version: string
+  platform: string
+  arch: string
+  start: any
+  config: Config
+
+  constructor (command: string, plugin: string, pluginVersion: string, config: Config) {
+    this.config = config
+    this.command = command
+    this.plugin = plugin
+    this.pluginVersion = pluginVersion
+    this.platform = OS.platform()
+    this.arch = OS.arch()
+  }
+
+  get version (): string {
```

Comment:
I might be missing something but this doesn't seem necessary since you could just populate the field in the constructor like you do for other properties